### PR TITLE
Settings Modal V3

### PR DESCRIPTION
**Pull request by Mi-Lan**
- We reopened this pr since it was causing which went unnoticed https://github.com/protofire/omen-exchange/issues/2083

Old PR info:
This pr addresses issue where rpc node is down and displays appropriate settings modal where you can choose different rpc url, here are steps to test it out:

1) Go to src/util/networks.ts
2) If you want to test the default url breaking go to networks object and fuckup the url and first alternative url for appropriate network you are on
Screenshot 2021-07-06 at 13 53 11
3) If you wan't to test the alternative url thats already set in localStorage, inspect in chrome go to Application tab and then to localStorage there you will find rpcAddress and just fuck up the url and refresh the page 
Screenshot 2021-07-06 at 13 55 33

Check list
- [ ] Changing of rpc url in your connection modal displays urls for appropriate network user is on and changes them appropriately
- [ ] When rpc url is broken modal should popup with option to select alternative url for the network user is on and when changed to working rpc it should reload page and not show modal anymore. Also If we are on xDai tight/relayer it should detect and show xDai alternative urls

### CODE REVIEW

**Review by hexyls — APPROVED** [review]

File: app/src/util/connectors.ts [hunk 1]
```diff
@@ -88,7 +88,8 @@ class WalletConnectConnector extends Connectors.Connector {
 const WalletConnect = new WalletConnectConnector()
 
 const Infura = new NetworkOnlyConnector({
-  providerURL: infuraNetworkURL,
+  providerURL:
+    localStorage.getItem('relay') === 'true' ? networks[networkIds.XDAI].url : networks[networkIds.MAINNET].url,
```
Comment (Mi-Lan) [inline app/src/util/connectors.ts:92]: @hexyls Hey men I added this as well I was having some issues where incorrect url was being checked and modal wasn't showing up...and this fixed it. What do you think?

**Comment by pimato** [comment]
Check list
- [ ] Changing of rpc url in your connection modal displays urls for appropriate network user is on and changes them appropriately
- [ ] When rpc url is broken modal should popup with option to select alternative url for the network user is on and when changed to working rpc it should reload page and not show modal anymore. Also If we are on xDai tight/relayer it should detect and show xDai alternative urls

**Review by pimato — APPROVED** [review]

**Comment by KadenZipfel** [comment]
If I mess up the rpc address, the modal pops up but I can't save with the currently selected rpc, would make perfect sense if I could simply hit save

image

**Review by KadenZipfel — APPROVED** [review]
LGTM!

File: app/src/util/connectors.ts (hunk 1 above)
Comment (hexyls) [reply app/src/util/connectors.ts:92]: looks good 👍

**Review by hexyls — APPROVED** [review]
🙌

**Comment by Mi-Lan** [comment]
> If I mess up the rpc address, the modal pops up but I can't save with the currently selected rpc, would make perfect sense if I could simply hit save. Super low priority though so would be chill to move it into a new issue
> 
> image

@KadenZipfel Yeah I can remove that easily! I think geronimo suggested that when I made it the first time...

**Pull request merged**